fiftyone/core/dataset.py

```diff
@@ -1626,6 +1644,526 @@ def add_frame_field(
+                summary data. By default, a suitable name is derived from the
+                given ``path``
+            sidebar_group (None): the name of a
+                :ref:`App sidebar group <app-sidebar-groups>` to which to add
+                the summary field, if necessary. By default, all summary fields
+                are added to a ``"summaries"`` group. You can pass ``False`` to
+                skip sidebar group modification
+            include_counts (False): whether to include per-value counts when
+                summarizing categorical fields
+            group_by (None): an optional attribute to group by when ``path``
+                is a numeric field to generate per-attribute ``[min, max]``
+                ranges. This may either be an absolute path or an attribute
+                name that is interpreted relative to ``path``
+            read_only (True): whether to mark the summary field as read-only
+            create_index (True): whether to create database index(es) for the
+                summary field
+            overwrite (True): whether to overwrite any existing summary field
+                with the same name
+
+        Returns:
+            the summary field name
+        """
+        _field = self.get_field(path)
+
+        if isinstance(_field, (fof.StringField, fof.BooleanField)):
+            index_type = "categorical"
+        elif isinstance(
+            _field,
+            (fof.FloatField, fof.IntField, fof.DateField, fof.DateTimeField),
+        ):
+            index_type = "numeric"
+        elif _field is not None:
+            raise ValueError(
+                f"Cannot generate a summary for field '{path}' of "
+                f"type {type(_field)}"
+            )
+        else:
+            raise ValueError(
+                "Cannot generate a summary field for non-existent or "
+                f"undeclared field '{path}'"
+            )
+
+        _path, is_frame_field, list_fields, _, _ = self._parse_field_name(path)
+
+        if field_name is None:
+            _chunks = _path.split(".")
+
+            chunks = []
+            if is_frame_field:
+                chunks.append("frames")
+
+            found_list = False
+            for i, _chunk in enumerate(_chunks, 1):
+                if ".".join(_chunks[:i]) in list_fields:
+                    found_list = True
+                    break
+                else:
+                    chunks.append(_chunk)
+
+            if found_list:
+                chunks.append(_chunks[-1])
+
+            field_name = "_".join(chunks)
+
+        field = self.get_field(field_name)
+        if field is not None:
+            if overwrite and _SUMMARY_FIELD_KEY in field.info:
+                self.delete_summary_field(field_name)
+            else:
+                raise ValueError(f"Field '{field_name}' already exists")
+
+        index_fields = []
+        summary_info = {"path": path, "index_type": index_type}
```

Comment:
probably should rename this into `field_type`

Reply:
IDK personal preference 🤷

Reply:
I think the main confusing point for me is that we are not creating an actual db index, but a summary field, so `index_type` is a little bit confusing if someone else has to read it again in a couple of months.

Reply:
oh I thought you wanted `summary_info -> field_type`. You want `index_type -> field_type`. Yeah I agree with that, will update

Reply:
done ✅